presto-main/src/main/java/com/facebook/presto/type/Re2JRegexp.java

```diff
@@ -88,6 +91,45 @@ public Slice replace(Slice source, Slice replacement)
         }
     }
 
+    public Slice replace(Slice source, UnaryFunctionInterface replaceFunction)
+    {
+        Matcher matcher = re2jPattern.matcher(source);
+        try {
+            matcher.reset();
+            int appendPos = 0;
+            SliceOutput sliceOutput = new DynamicSliceOutput(source.length());
+            while (matcher.find()) {
+                int s = matcher.start();
+                int e = matcher.end();
```

Comment:
I think we need to handle empty matches similar as for Joni

Reply:
If you're talking about the "nextStart = matcher.getEnd() + 1;" case for Joni, RE2J handles the updating of start position internally in Matcher.find() for empty match, whereas in Joni it only provides search(int start, int range, int option) and we need to calculate start position ourselves.

Reply:
indeed, +1